crates/language/src/language_settings.rs

```diff
@@ -44,6 +44,7 @@ pub struct AllLanguageSettings {
     pub copilot: CopilotSettings,
     defaults: LanguageSettings,
     languages: HashMap<Arc<str>, LanguageSettings>,
+    pub associations: HashMap<Arc<str>, Vec<Arc<str>>>
```

Comment:
What happens on the corresponding settings change?
We need to do a lot of things when language associations change: reload+reparse all the buffers open, repopulate the semantic index, restart (and kill the LSP servers), ...

Some of that might work already, but all of it needs more tests written.
We also need to ensure that non-corresponding settings change does not do all this.

Reply:
>We need to do a lot of things when language associations change: reload+reparse all the buffers open, repopulate the semantic index, restart (and kill the LSP servers), ...

Is this the function [`on_settings_changed(&mut self, cx: &mut ModelContext<Self>)`](https://github.com/zed-industries/zed/blob/ac74a72a9ef80a6cc50e7b001e163e76fbd9fa3d/crates/project/src/project.rs#L845) that requires carefully update? It seems that most buffer-related things on settings change are handled within it.

I will add more tests for them after completing the basic functionalities.

Reply:
Yes, that's the one and I would assume something might work out of the box because we react to buffer language changes, but we have to prove it via testing.